tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,173 @@
+
+## What is hspec-golden?
+[hspec-golden](https://github.com/stackbuilders/hspec-golden) is a superset for the popular Haskell B.D.D. testing framework [hspec](https://github.com/hspec/hspec),
+that will allow us to create golden test for our Haskell projects,
+currently being developed by [Cristhian Motoche](https://github.com/CristhianMotoche) an active Open Source code enthusiast and Stackbuilders developer.
+
+## What do we need?
+For this tutorial, we need an environment running Haskell with base 4.6 up to 5,
+if we don’t have a Haskell environment we could refer to this [link](https://wiki.haskell.org/Haskell_in_5_steps#Install_Haskell) to get a clean installation using `stack` or `cabal`.
+
+We also need to have added as a dependency to our Haskell project the [hspec](https://github.com/hspec/hspec) testing framework.
+
+With stack:
+
+``$ stack install hspec``
+
+With cabal:
+
+``$ cabal update && cabal install hspec``
+
+## Getting started
+This code sample will help us to understand how does basic golden testing works.
+
+## Installing the library:
+To install the [hspec-golden](https://github.com/stackbuilders/hspec-golden) superset we need to run this command on the console:
+
+``$ stack install hspec-golden``
+
+## Importing the libraries
+In our project on we have created a GoldenSpec.hs and added the following lines.
+``` 
+module GoldenSpec where
+
+import Test.Hspec
+import Test.Hspec.Golden
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as BL
+import Data.Yaml as Y
+import Data.Aeson as A
+import Control.Exception (evaluate)
+```
+
+Here we are telling Haskell to add the libraries that are going to be implemented in this tutorial,
+we are adding the test framework [hspec](https://github.com/hspec/hspec) and [hspec-golden](https://github.com/stackbuilders/hspec-golden),
+with some additional libs to implement our own Golden type.
+
+
+### Writing our first golden test
+The [hspec-golden](https://github.com/stackbuilders/hspec-golden) library provides us with an easy example to test the functionality of the library with the `defaultGolden` helper:
+
+``` 
+describe "id" $
+     it "generates the right output with the right params" $
+        let output = show $ id (1:: Int)
+        in defaultGolden "id" output
+```
+
+On this test we are setting up a B.D.D. test with [hspec](https://github.com/hspec/hspec) “describe” and “it”,
+but on the inside there is an implementation of the `defaultGolden` helper,
+that will create(or update) for us a directory called `.golden` this will contain another directory named after our described test in this case `id` with two files containing the `actual` and `golden` values for our test.
+
+For our example we will create our own Golden type.
+We have created a file called test.yml that contains an ansible role on the dir ["test/fixtures/"](https://github.com/MedAdrian/hspec-golden-tutorial/blob/master/test/fixtures/test.yml) 
```

Comment:
Is the plan to move this repository to the Stack Builders organization?

Is Ansible necessary?

Reply:
Should the example be under SB as a private repo? or just as a code folder inside the tutorial?

Reply:
I think either way works, under Stack Builders as a public repository if you're linking from the tutorial, or inside the tutorials repository (I think other tutorials do this).